crates/router/src/core/payments/transformers.rs

```diff
@@ -4101,10 +4101,10 @@ impl ForeignFrom<(storage::PaymentIntent, Option<storage::PaymentAttempt>)>
 {
     fn foreign_from((pi, pa): (storage::PaymentIntent, Option<storage::PaymentAttempt>)) -> Self {
         Self {
-            id: pi.id,
-            merchant_id: pi.merchant_id,
-            profile_id: pi.profile_id,
-            customer_id: pi.customer_id,
+            id: pi.id.clone(),
```

Comment:
These clones are required because `is_split_payment` function borrows PaymentIntent and without these `clone()` calls, there is an error, `Borrow of Partially moved vaule`